editor/filesystem_dock.cpp

```diff
@@ -1283,42 +1283,71 @@ void FileSystemDock::_try_duplicate_item(const FileOrFolder &p_item, const Strin
 	}
 
 	Ref<DirAccess> da = DirAccess::create(DirAccess::ACCESS_RESOURCES);
-	print_verbose("Duplicating " + old_path + " -> " + new_path);
-	Error err = p_item.is_file ? da->copy(old_path, new_path) : da->copy_dir(old_path, new_path);
-	if (err == OK) {
-		// Move/Rename any corresponding import settings too.
-		if (p_item.is_file && FileAccess::exists(old_path + ".import")) {
-			err = da->copy(old_path + ".import", new_path + ".import");
+
+	if (p_item.is_file) {
+		print_verbose("Duplicating " + old_path + " -> " + new_path);
+
+		// Create the directory structure
+		da->make_dir_recursive(new_path.get_base_dir());
+
+		if (FileAccess::exists(old_path + ".import")) {
+			Error err = da->copy(old_path, new_path);
 			if (err != OK) {
-				EditorNode::get_singleton()->add_io_error(TTR("Error duplicating:") + "\n" + old_path + ".import\n");
+				EditorNode::get_singleton()->add_io_error(TTR("Error duplicating:") + "\n" + old_path + ": " + error_names[err] + "\n");
+				return;
 			}
 
 			// Remove uid from .import file to avoid conflict.
 			Ref<ConfigFile> cfg;
 			cfg.instantiate();
-			cfg->load(new_path + ".import");
+			cfg->load(old_path + ".import");
 			cfg->erase_section_key("remap", "uid");
-			cfg->save(new_path + ".import");
-		} else if (p_item.is_file && (old_path.get_extension() == "tscn" || old_path.get_extension() == "tres")) {
-			// FIXME: Quick hack to fix text resources. This should be fixed properly.
-			Ref<FileAccess> file = FileAccess::open(old_path, FileAccess::READ, &err);
-			if (err == OK) {
-				PackedStringArray lines = file->get_as_utf8_string().split("\n");
-				String line = lines[0];
-
-				if (line.contains("uid")) {
-					line = line.substr(0, line.find(" uid")) + "]";
-					lines.write[0] = line;
+			err = cfg->save(new_path + ".import");
+			if (err != OK) {
+				// Delete the duplicated file
+				da->remove(new_path);
```

Comment:
Why delete the file? If import fails to save, it will simply be reimported.

Reply:
This is true, but it will probably be reimported with different settings. Therefore I decided to delete the file.

Reply:
Well, if you duplicate a file, it's better to have it reimported with different settings than not getting duplicated at all.
Not to mention such situation is rare anyway.